RogueMaster/flipperzero-firmware-wPlugins
# Added wallet storage capability

# What's new

### Saving wallets to SD card

  - Wallets are saved to SD card upon creation in `apps_data/flipbip`
      - NOTE: `apps_data` folder must already exist on SD card!
  - Saved wallets can be viewed between app runs
  - Wallets are encrypted with a randomly generated key, and that key is also encrypted
      - `.flipbip.dat` and `.flipbip.key` files are both required to be in `apps_data/flipbip`
      - Backups of both these files `.flipbip.dat.bak` and `.flipbip.key.bak` are also maintained

# Verification 

- Compile and run on Flipper
- Select menu item in app for "Generate new wallet"
- Verify that there are now menu options to view BTC and ETH wallets in menu
- Verify that when viewing these wallets, they have the same mnemonic seed phrase
- Verify data files were created on SD card at `apps_data/flipbip/`

# Checklist (For Reviewer)
